OpenSDS_Volume_Actions_Design.md

```diff
@@ -0,0 +1,92 @@
+# Title
+
+Status: Pending
+
+Version: Beta
+
+Author(s): PengYiOfOpenSDS
+
+## Summary
+
+Extends the size of a volume. 
+
+## Motivation
+
+Extends the size of a volume to a requested size, in gibibytes (GiB). 
+
+### Goals
+
+What will be addressed by the proposal?
+Extends the size of a volume to a requested size, in gibibytes (GiB). 
+
+### Non-Goals
+
+What will not be addressed by the proposal?
+"Shrinks the size of a volume" will not be addressed.
+
+## Proposal
+
+Extends the size of a volume to a requested size, in gibibytes (GiB). 
+
+### Data model impact
+
+Does the proposal affect data model?
+No
+
+### REST API impact
+
+Does the proposal affect API?
+v1/{ProjectID}/ volumes/{VolumeID} /action
+Method: POST
+JSON schema definition:
+{
+"extend": {
+"newSize": 3
+}
+}
+Successful response code: 202
+
+### Security impact
+
+Does it affect security? 
+No
+
+### Other end user impact
+
+Other than API, are there other changes that affect a user? 
+Yes, add a cli command (osdsctl volume extend <id> <new size>).
+
+### Performance impact
+
+Is there any performance impact? 
+No
+
+### Other deployer impact
+
+How does it affect the deployment? 
+No
+
+Are there config option changes? 
+No
+
+### Developer impact
```

Comment:
what does the interface of driver look like？

Reply:
Can you please address this comment?